includes/class-wc-privacy-erasers.php

```diff
@@ -0,0 +1,315 @@
+	 * @return array An array of personal data in name value pairs
+	 */
+	public static function order_data_eraser( $email_address, $page ) {
+		$page            = (int) $page;
+		$user            = get_user_by( 'email', $email_address ); // Check if user has an ID in the DB to load stored personal data.
+		$erasure_enabled = wc_string_to_bool( get_option( 'woocommerce_erasure_request_removes_order_data', 'no' ) );
+		$response        = array(
+			'items_removed'  => false,
+			'items_retained' => false,
+			'messages'       => array(),
+			'done'           => true,
+		);
+
+		$order_query = array(
+			'limit'    => 10,
+			'page'     => $page,
+			'customer' => array( $email_address ),
+		);
+
+		if ( $user instanceof WP_User ) {
+			$order_query['customer'][] = (int) $user->ID;
+		}
+
+		$orders = wc_get_orders( $order_query );
+
+		if ( 0 < count( $orders ) ) {
+			foreach ( $orders as $order ) {
+				if ( apply_filters( 'woocommerce_privacy_erase_order_personal_data', $erasure_enabled, $order ) ) {
+					WC_Privacy::remove_order_personal_data( $order );
+
+					/* Translators: %s Order number. */
+					$response['messages'][]    = sprintf( __( 'Removed personal data from order %s.', 'woocommerce' ), $order->get_order_number() );
+					$response['items_removed'] = true;
+				} else {
+					/* Translators: %s Order number. */
+					$response['messages'][]     = sprintf( __( 'Retained personal data in order %s due to settings.', 'woocommerce' ), $order->get_order_number() );
+					$response['items_retained'] = true;
+				}
+			}
+			$response['done'] = 10 > count( $orders );
+		} else {
+			$response['done'] = true;
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Finds and exports customer download logs by email address.
+	 *
+	 * @since 3.4.0
+	 * @param string $email_address The user email address.
+	 * @param int    $page  Page.
+	 * @return array An array of personal data in name value pairs
+	 */
+	public static function download_data_eraser( $email_address, $page ) {
+		$page            = (int) $page;
+		$user            = get_user_by( 'email', $email_address ); // Check if user has an ID in the DB to load stored personal data.
+		$erasure_enabled = wc_string_to_bool( get_option( 'woocommerce_erasure_request_removes_download_data', 'no' ) );
+		$response        = array(
+			'items_removed'  => false,
+			'items_retained' => false,
+			'messages'       => array(),
+			'done'           => true,
+		);
+
+		$downloads_query = array(
+			'limit'  => -1,
+			'page'   => $page,
+			'return' => 'ids',
+		);
+
+		if ( $user instanceof WP_User ) {
+			$downloads_query['user_id'] = (int) $user->ID;
+		} else {
+			$downloads_query['user_email'] = $email_address;
+		}
+
+		$customer_download_data_store     = WC_Data_Store::load( 'customer-download' );
+		$customer_download_log_data_store = WC_Data_Store::load( 'customer-download-log' );
+		$downloads                        = $customer_download_data_store->get_downloads( $downloads_query );
```

Comment:
`$downloads` is not used anywhere in this method.

Should this method also delete stuff from the download logs? I wasn't able to successfully run the removal tool (see comment below) but I don't see a mechanism for it here. It seems like we should delete download logs for the email (and user IP if possible).

<img width="1946" alt="screen shot 2018-04-30 at 3 12 35 pm" src="https://user-images.githubusercontent.com/7317227/39452944-f5ff8548-4c88-11e8-9992-82a96d3cfb77.png">

Reply:
This is managed by the DB schema now - we added some foreign keys.